packages/@aws-cdk/cdk/lib/cfn-resource.ts

```diff
@@ -205,33 +193,21 @@ export class CfnResource extends CfnRefElement {
    */
   public _toCloudFormation(): object {
     try {
-      // merge property overrides onto properties and then render (and validate).
-      const tags = TagManager.isTaggable(this) ? this.tags.renderTags() : undefined;
-      const properties = deepMerge(
-        this.properties || {},
-        { tags },
-        this.untypedPropertyOverrides
-      );
-
       const ret = {
         Resources: {
           // Post-Resolve operation since otherwise deepMerge is going to mix values into
           // the Token objects returned by ignoreEmpty.
           [this.logicalId]: new PostResolveToken({
             Type: this.resourceType,
-            Properties: ignoreEmpty(properties),
+            Properties: ignoreEmpty(this.renderProperties()),
             DependsOn: ignoreEmpty(renderDependsOn(this.dependsOn)),
-            CreationPolicy:  capitalizePropertyNames(this, this.options.creationPolicy),
+            CreationPolicy: capitalizePropertyNames(this, this.options.creationPolicy),
             UpdatePolicy: capitalizePropertyNames(this, this.options.updatePolicy),
             UpdateReplacePolicy: capitalizePropertyNames(this, this.options.updateReplacePolicy),
             DeletionPolicy: capitalizePropertyNames(this, this.options.deletionPolicy),
             Metadata: ignoreEmpty(this.options.metadata),
             Condition: this.options.condition && this.options.condition.logicalId
           }, props => {
```

Comment:
Curious why we moved renderProperies outside of PostResolve. I suspect the reasoning hasn’t changed or am I missing something?

Reply:
Also feel like it should be inside.

Reply:
moved back, but had to re-separate properties and validator/mapper.